### DISCUSSION TITLE
tr/integrations/ray-tune/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#16009] [Docs]
# tr/integrations/ray-tune/

Optimize YOLOv8 Ray Tune ile model performansı. Gelişmiş arama stratejilerini, paralelliği ve erken durdurmayı kullanarak verimli hiper parametre ayarlamayı öğrenin.

https://docs.ultralytics.com/tr/integrations/ray-tune/

**Comment by skyceeee** [comment]
from ray import tune

from ultralytics import YOLO

model = YOLO("yolov8n.pt")
search_space = {"lr0": tune.uniform(1e-5, 1e-1), "momentum": tune.uniform(0.6, 0.98)}
result_grid = model.tune(data="coco8.yaml", space=search_space, use_ray=True)
------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-82-78e91581240d>](https://localhost:8080/#) in <cell line: 7>()
      5 model = YOLO("yolov8n.pt")
      6 search_space = {"lr0": tune.uniform(1e-5, 1e-1), "momentum": tune.uniform(0.6, 0.98)}
----> 7 result_grid = model.tune(data="coco8.yaml", space=search_space, use_ray=True)

/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py in tune(self, use_ray, iterations, *args, **kwargs)

ModuleNotFoundError: No module named 'ultralytics.utils.tuner'


why am i getting this error?

**Reply by glenn-jocher** [reply]
@skyceeee it seems you're encountering a `ModuleNotFoundError`. Please ensure you have the latest version of the Ultralytics package installed. You can update it using `pip install -U ultralytics`. If the issue persists, let us know!

**Reply by skyceeee** [reply]
I did what you said and my ultralytics version is 8.2.87 but I still encounter the same error.

**Reply by skyceeee** [reply]
It works on code vs code, but I encounter that error in colab.

**Reply by glenn-jocher** [reply]
It seems you're encountering this error in Google Colab. Please ensure that all necessary packages are installed in your Colab environment. You can do this by running `!pip install -U ultralytics "ray[tune]"`. If the issue persists, try restarting the runtime. Let me know if you need further assistance!
